.gitmodules

```diff
@@ -1,3 +1,3 @@
-[submodule "prediction_market_agent/agents/microchain_agent/mech/mech"]
-	path = prediction_market_agent/agents/microchain_agent/mech/mech
+[submodule "prediction_market_agent/tools/mech/mech"]
```

Comment:
Why is it advised to use submodules vs poetry dependencies?

Reply:
Ahh, when I first did this I thought that [mech](https://github.com/valory-xyz/mech) wasn't configured to be installed as a python package, but I am wrong!

It isn't released on pypi, but it is possible to install by adding `mech = { git = "https://github.com/valory-xyz/mech.git", branch = "main" }` to the pyproject.toml file.

Reply:
I tried switching to this, and it's not a clean install due to dependency issues. Spent an hour trying to fix these, but no luck!

With changes to [PMA](https://github.com/gnosis/prediction-market-agent/pull/95) and [PMAT](https://github.com/gnosis/prediction-market-agent-tooling/pull/213) dependencies I was blocked by an unresolvable conflict with crewai.

With changes the other way to the [mech](https://github.com/valory-xyz/mech/pull/218) repo I was blocked by conflicts with the mech-client repo (required by PMAT).

Reply:
Submodule is a workaround for this for now... but I'll make a ticket to fix the conflicts and replace the submodule with a package dependency.

Reply:
issue here: https://github.com/gnosis/prediction-market-agent/issues/97